Apromore-Extras/APMLogModule/src/main/java/org/apromore/apmlog/LaLog.java

```diff
@@ -308,22 +302,20 @@ public void setVariantIdFreqMap(UnifiedMap<Integer, Integer> variantIdFreqMap) {
 
     @Override
     public double getMinDuration() {
-        return minDuration;
+        return !caseDurationList.isEmpty() ? caseDurationList.min() : 0;
     }
 
     @Override
     public void setMinDuration(double minDuration) {
-        this.minDuration = minDuration;
     }
 
     @Override
     public double getMaxDuration() {
-        return maxDuration;
+        return !caseDurationList.isEmpty() ? caseDurationList.max() : 0;
     }
 
     @Override
     public void setMaxDuration(double maxDuration) {
-        this.maxDuration = maxDuration;
     }
```

Comment:
These small changes look fine for me, changing from using min, max variables to using a list of case durations. Can we add unit tests to cover the affected public methods in this PR (I didn't find these unit tests)?
- Unit tests to cover getMinDuration, getMaxDuration, getAverageDuration, getMedianDuration for each log class: LaLog (LaLogTest), PLog (PLogTest), ImmutableLog (ImmutableLogTest), 
- Unit test for LogFactory (LogFactoryTest) to cover convertXLog method)
- Unit test for ImmutableTrace (ImmutableTraceTest) to cover getStartTimeMili and getEndTimeMilli methods.